test/acceptance/user-order.controller.acceptance.ts

```diff
@@ -58,7 +58,7 @@ describe('UserOrderController acceptance tests', () => {
 
     expect(res.body.orderId).to.be.a.String();
     delete res.body.orderId;
-    expect(res.body).to.deepEqual(order);
+    expect(res.body).to.deepEqual(JSON.parse(JSON.stringify(order)));
```

Comment:
`order.toJSON()` should do the trick, isn't it?

Reply:
IIRC, `order.toJSON()` returns an object that includes properties with undefined values.

```
// our expectation
{
  title: 'My Todo',
  isCompleted: undefined
}
```

On the other hand, the actual JSON string send by the server and parsed by the client in the test leaves out properties set to `undefined`. 

```
// the actual value
{
  title: 'My Todo'
}
```

AFAIK, the deep equality check distinguishes between a property set to `undefined` and a property that does not exist on the object at all.

As a result, the expected object obtained via `.toJSON()` is  rejected as being different from the actual value parsed from a JSON string.

I have encountered this issue myself in https://github.com/strongloop/loopback-next/pull/1679, I am going to add a new `testlab` helper to deal with that. See https://github.com/strongloop/loopback-next/blob/b4bba3d09d663b2919bd6ca3c5a5430d70e1d8c7/packages/testlab/src/misc.ts

Based on this discussion, I am going to open a new pull request to add this helper only, so that we can start using it sooner.

Reply:
> Based on this discussion, I am going to open a new pull request to add this helper only, so that we can start using it sooner.

See https://github.com/strongloop/loopback-next/pull/1733

Reply:
See also https://github.com/strongloop/loopback-next/pull/1734